python/GafferUI/ColorChooser.py

```diff
@@ -817,12 +808,18 @@ def __init__( self, color=imath.Color3f( 1 ), **kw ) :
 					self.__spacers["hsv"] = GafferUI.Spacer( imath.V2i( 0, spacerHeight ), parenting = {  "index" : ( 0, 5 ) } )
 					self.__spacers["tmi"] = GafferUI.Spacer( imath.V2i( 0, spacerHeight ), parenting = { "index" : ( 0, 9 ) } )
 
-				# Options Button
-				GafferUI.MenuButton(
-					image = "gear.png",
-					menu = GafferUI.Menu( Gaffer.WeakMethod( self.__optionsMenuDefinition ) ),
-					hasFrame = False
-				)
+					# Options Button.
+
+					GafferUI.Spacer( imath.V2i( 4, 0 ), parenting = { "index" : ( 6, slice( 0, grid.gridSize().y ) ) } )
+					GafferUI.MenuButton(
+						image = "gear.png",
+						menu = GafferUI.Menu( Gaffer.WeakMethod( self.__optionsMenuDefinition ) ),
+						hasFrame = False,
+						# Occupy an entire column of the grid, but aligned to the top of the column.
+						# This keeps the button aligned with the top of the first visible slider, no
+						# matter which that is.
+						parenting = { "index" : ( 7, slice( 0, grid.gridSize().y ) ), "alignment" : ( GafferUI.HorizontalAlignment.None_, GafferUI.VerticalAlignment.Top ) }
```

Comment:
This is what I did to keep the options button where we want it, while keeping _ColorFieldRowLayout at a size of 2 widgets. I was worried that the alignment would trigger some more weird Qt behaviour, but it seems to be OK.

Reply:
That does feel nicer to have it work with any slider configuration.